internal/reloader/reloader.go

```diff
@@ -105,17 +103,18 @@ func (rl *Reloader) Watch(ctx context.Context, fn ConfigFunc) error {
 
 			cfg, err := rl.reload()
 			if err != nil {
-				logrus.WithError(err).Error("config reload failed")
+				logrus.WithError(err).Error(err)
 				continue
 			}
 
 			// Call the callback function with the latest cfg.
 			fn(cfg)
 
-		case evt, ok := <-wr.Events:
+		case evt, ok := <-wr.Events():
 			if !ok {
-				logrus.WithError(err).Error("fsnotify has exited")
-				return nil
+				err := errors.New("fsnotify event channel was closed")
+				logrus.WithError(err).Error("fsnotify event channel was closed")
```

Comment:
Ideally prefix the errors with the package name, sometimes it's super hard to track down certain strings.

Reply:
Fixed